manifests/config.pp

```diff
@@ -10,13 +10,72 @@
     system => true,
   }
 
+  case $facts['os']['family'] {
+    'RedHat':  {
+      $nologin_shell = '/sbin/nologin'
+      case $facts['os']['release']['major'] {
+        '7': {
+          systemd::unit_file { 'caddy.service':
```

Comment:
instead of matching on the OS versions, would it make sense to match on a fact for the service_provider? (that one already exists). There are a bunch of distros in the world that use systemd, that might make the file shorter and easier to adopt new OSes.

Reply:
Very good point, for the manifest. Might need to be manually set in rspec since I am not sure that fact is provided by facterdb, but will test it.